app/controllers/admin/playlist_activities_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Admin::PlaylistActivitiesController < Admin::ResourceController
+  def new
+    @student = Student.find(params[:student_id])
+    @resource = @student.playlist.playlist_activities.new
+    authorize(resource)
+    respond_with resource
+  end
+
+  private
+    def resource_as_sym
+      :playlist_activity
+    end
+
+    def whitelist
+      [:activity_id, :focus_area_id]
+    end
+
+    def build_resource
+      @student = Student.find(params[:student_id])
+      @resource = @student.playlist.playlist_activities.new(permitted_params)
+      @resource.position = (@student.playlist.playlist_activities.maximum(:position) || 0) + 1
```

Comment:
if this is to handle sorting, and the ui portion is not done yet, lets hold off on the sorting part and implement this as an Ajax request on drag and drop in the list. Then we only need to care about the currently focused activity and the one you're putting it in front of, so you just need to discretely update two columns.